scripts/test-puppeteer.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+set -eo pipefail
+
+# Function to stop the browserless docker container
+stop_docker_container() {
+    if [ -z "$GITHUB_ACTIONS" ] && [ -n "$CONTAINER_ID" ]; then
+        echo "Stopping browserless docker container..."
+        docker stop $CONTAINER_ID
+    fi
+}
+
+# Function to stop the development server
+stop_dev_server() {
+    if [ -z "$GITHUB_ACTIONS" ] && [ -n "$DEV_SERVER_PID" ]; then
+        echo "Stopping development server..."
+        kill $DEV_SERVER_PID
+        wait $DEV_SERVER_PID 2>/dev/null
+    fi
+}
+
+cleanup() {
+    stop_dev_server
+    stop_docker_container
+}
+
+# Set up trap to call cleanup function on script exit or if the program crashes
+trap cleanup EXIT INT TERM
+
+# Check if we're running in a GitHub Action
+if [ -z "$GITHUB_ACTIONS" ]; then
+    # We're not in a GitHub Action, so start the browserless docker container
+    echo "Starting browserless docker container..."
+    CONTAINER_ID=$(docker run -d --rm -p 7566:3000 -e "CONNECTION_TIMEOUT=-1" browserless/chrome)
+
+    # Wait for the container to be ready
+    echo "Waiting for browserless to be ready..."
+    sleep 5
+fi
+
+# Check if we're not in GitHub Actions
+if [ -z "$GITHUB_ACTIONS" ]; then
```

Comment:
This checks if we ARE running in a GitHub Action (it's the same condition as above)

Reply:
Oops, I meant ARE NOT. You got it:).